fio_wrapper/trigger_fio.py

```diff
@@ -79,44 +79,51 @@ def _log_payload(self, directory, user, uuid, sample, fio_jobs_dict, fio_version
+                for numjob in numjob_list:
+                    log_file_prefix_string = 'write_' + str(log) + '_log'
+                    if log in ['clat','slat']:
+                        log_file_prefix_string = 'write_lat_log'
                     try:
-                        log_file_name = str(fio_jobs_dict['global'][log_file_prefix_string]) + '_' + str(log) + '.1.log.' + str(host)
-                    except:
-                        logger.info("Error setting log_file_name")
-                with open(directory+'/'+str(log_file_name), 'r') as log_file:
-                    for log_line in log_file:
-                        log_line_values = str(log_line).split(", ")
-                        if len(log_line_values) == 4:
-                            timestamp_ms = int(fio_starttime[host]) + int(log_line_values[0])
-                            newtime = datetime.fromtimestamp(timestamp_ms / 1000.0)
-                            log_dict = {
-                              "uuid": uuid,
-                              "user": user,
-                              "host": host,
-                              "cluster_name": self.cluster_name,
-                              "fio-version": fio_version,
-                              "job_options": job_options,
-                              "job_name": str(job),
-                              "sample": int(sample),
-                              "log_name": str(log),
-                              "timestamp": timestamp_ms, #this is in ms
-                              "date": newtime.strftime('%Y-%m-%dT%H:%M:%S.%fZ'),
-                              str(_current_log_files[log]['metric']): int(log_line_values[1]),
-                              #"nodeName": pod_details["hostname"],
-                              "data_direction": _data_direction[int(log_line_values[2])],
-                              "offset": int(log_line_values[3])
-                            }
-                            if 'global' in fio_jobs_dict.keys():
-                                log_dict['global_options'] = fio_jobs_dict['global']
-                            logs.append(log_dict)
+                        log_file_name = str(job_options[log_file_prefix_string]) + '_' + str(log) + '.' + numjob + '.log.' + str(host)
+                    except KeyError:
+                        try:
+                            log_file_name = str(fio_jobs_dict['global'][log_file_prefix_string]) + '_' + str(log) + '.' + numjob + '.log.' + str(host)
+                        except:
+                            logger.info("Error setting log_file_name")
+                    with open(directory+'/'+str(log_file_name), 'r') as log_file:
+                        for log_line in log_file:
+                            log_line_values = str(log_line).split(", ")
+                            if len(log_line_values) == 4:
+                                timestamp_ms = int(fio_starttime[host]) + int(log_line_values[0])
```

Comment:
not sure if this is right always, this means we're assuming that all the jobs start at the exact same moment. have to test it to make sure its right, i'd seen it wasn't the case

Reply:
Can you elaborate on where you see the problem? The line your comment is on is unchanged from current code, as is the entire `try` here except for replacing the `1` with `str(numjob)`, AFAICT.

Reply:
I'm referring to the timestamp which is the exact line I commented on, the individual log files don't have absolute timestamp, here we are using the start time found in the result json document and assuming it'll be the same for all the jobs( hence the line hasn't changed ). A little concerned about it, but not a blocker if we're all in agreement that this assumption isn't a big problem :)

Reply:
I don't believe this is an issue because fio client is the one writing these logs not separate fio instances, I assume the client at X "time past start time" writes down how many iops(or other metric) were produced for each server and job. that would be based on start time and if some servers/jobs were delayed they would have a lower iops count at X "time past start time". at the end of the job the client would then issue a stop command and all instance would die. as of right now the data supports this.